### PR TITLE
Patch for Issue #301

### PR DESCRIPTION
**Pull request by radamanthus**
Download Redis installer in /opt. 

https://github.com/engineyard/ey-cloud-recipes/issues/301

### CODE REVIEW

File: cookbooks/redis/recipes/default.rb [hunk 1]
```diff
@@ -17,27 +17,27 @@
...
       backup 0
-      not_if { FileTest.exists?("/data/#{redis_source_directory}") }
+      not_if { FileTest.exists?(redis_installer_directory) }
     end
 
-    execute "unarchive Redis source" do
-      command "cd /data && tar zxf redis-#{redis_version}.tar.gz && sync"
-      not_if { FileTest.directory?("/data/#{redis_source_directory}") }
+    execute "unarchive Redis installer" do
+      command "cd /opt && tar zxf redis-#{redis_version}.tar.gz && sync"
```
Comment (AlexMorreale) [inline cookbooks/redis/recipes/default.rb]: In Chef it's generally better to use the `cwd '/opt/my/dir` then to do a `cd /opt/my/dir && cmd`

```
execute "unarchive Redis installer" do
  cwd "/opt"
  command "tar zxf redis-#{redis_version}.tar.gz && sync"
  ...
end

File: cookbooks/redis/recipes/default.rb [hunk 2]
```diff
@@ -17,27 +17,27 @@
...
-      command "mv /data/redis-#{redis_version} /data/redis-source"
-      not_if { FileTest.directory?("/data/redis-source") }
+    execute "rename /opt/redis-#{redis_version} to /opt/redis-source" do
+      command "mv /opt/redis-#{redis_version} #{redis_installer_directory}"
+      not_if { FileTest.directory?(redis_installer_directory) }
     end
 
     execute "run redis-source/make install" do
-      command "cd /data/redis-source && make install"
+      command "cd #{redis_installer_directory} && make install"
```
Comment (AlexMorreale) [inline cookbooks/redis/recipes/default.rb]: same as previous comment in terms of using `cwd` vs `cd /some/where && cmd`

**Comment by AlexMorreale** [comment]
Just a general thought but this will never install again as long as you just have that redis_source_directory. Won't that lead to false positives on the not_ifs if you ever delete the redis source or that dir. Wouldn't it be better to test for another file in the system like the redis-server binaries?

File: cookbooks/redis/recipes/default.rb (hunk 2 above)
Comment (radamanthus) [inline cookbooks/redis/recipes/default.rb]: Thanks for the feedback! I've updated the branch.

**Comment by radamanthus** [comment]
Completely agree on the concern for false positives. I just have to keep this PR short and sweet, that might have to be addressed in another PR.

**Comment by dvalfre** [comment]
Thanks @AlexMorreale, have opened #303 for further addressing the false positives.

**Comment by AlexMorreale** [comment]
Awesome!!! Happy to help/contribute.

**Pull request merged**